src/components/Button/PButton.vue

```diff
@@ -88,208 +78,172 @@
   })
 
   const classes = computed(() => ({
-    'p-button--primary': buttonClass.value === 'primary',
-    'p-button--secondary': buttonClass.value === 'secondary',
-    'p-button--inset': buttonClass.value === 'inset',
-    'p-button--flat': buttonClass.value === 'flat',
-    'p-button--danger': buttonClass.value === 'danger',
-    'p-button--danger--secondary': buttonClass.value === 'danger-secondary',
-    'p-button--rounded': props.rounded,
-    'p-button--equal-padding': props.icon && !slots.default,
-    'p-button-xs': props.size === 'xs',
-    'p-button-sm': props.size === 'sm',
-    'p-button-md': props.size === 'md',
-    'p-button-lg': props.size === 'lg',
-    'p-button-xl': props.size === 'xl',
+    'p-button--icon-only': props.icon && !slots.default,
+    'p-button--primary': props.primary,
+    'p-button--danger': props.dangerous,
+    'p-button--flat': props.flat,
+    'p-button--small': props.small,
+    'p-button--icon-prepend': props.icon && slots.default,
+    'p-button--icon-append': props.iconAppend && slots.default,
```

Comment:
These icon classes that check the slot. Whats the thought there?

Reply:
So the i guess it just keeps less classes applying. I'm realizing now, there's no harm in `icon-only` and either of these classes being applied at the same time since they all do the same thing (reduce left/right padding in some form). Feels a little cleaner to keep them? Or is dumping the extra checks and `useSlots` method call not worth?

Reply:
ah I see. I'm not sure `slots.default` will be reactive like you think. But this is fine for now. We can address if it becomes an issue.